src/compile/mark/bar.ts

```diff
@@ -17,6 +17,10 @@ export namespace bar {
 
     const stack = model.stack();
```

Comment:
@ayush29feb   Your question about stack is a very good one. 

In fact, the logic inside `mark/bar.ts` for stack should be correct here provided that the `stack` variable is correct. 
Thus, you have to prevent `stack` becomes truthy when `x2` or `y2` is specified, so that it doesn't go into that `if` clause.
You should to track down where this `stack` value comes from.  If you have questions, let me know.   
(Reading the docs about stack might help you understand when stack is turned on.)